packages/utils/src/lib/scoring.ts

```diff
@@ -77,27 +91,39 @@ export function calculateScore<T extends { weight: number }>(
   refs: T[],
   scoreFn: (ref: T) => number,
 ): number {
-  const numerator = refs.reduce(
-    (sum, ref) => sum + scoreFn(ref) * ref.weight,
-    0,
+  const { numerator, denominator } = refs.reduce(
+    (acc, ref) => {
+      acc.numerator += scoreFn(ref) * ref.weight;
+      acc.denominator += ref.weight;
+      return acc;
+    },
+    { numerator: 0, denominator: 0 },
   );
```

Comment:
Same here, please don't mutate the accumulator in `reduce` :pray: 

```suggestion
  const { numerator, denominator } = refs.reduce(
    (acc, ref) => ({
      ...acc,
      numerator: acc.numerator + scoreFn(ref) * ref.weight,
      denominator: acc.denominator + ref.weight,
    }),
    { numerator: 0, denominator: 0 },
  );
```

Reply:
![image](https://github.com/flowup/quality-metrics-cli/assets/148336720/a4ce46dd-d540-40ad-8791-2fc1a1612fb0)
Gone fully declarative ❤️